### PR TITLE
Add comilation instructions for MacOS

### PR DESCRIPTION
**Pull request by michaelbaudino**
This is a repost of https://github.com/fireice-uk/xmr-stak-cpu/pull/400.

This PR adds instructions to compile on MacOS.

### CODE REVIEW

**Comment by ExceptionallyGreat** [comment]
I have been successfully mining on my MacOS X with AMD GPU.

**Comment by michaelbaudino** [comment]
Thanks for your feedback @ExceptionallyGreat ❤️

I just updated the instructions with  Nvidia-related stuff. I used [Homebrew Cask](https://caskroom.github.io) to keep instructions simple (well, at least it's just one command) for beginners. Feel free to let me know what you think / what you'd change 😊

File: doc/compile_MacOS.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Compile **xmr-stak** for MacOS
+
+## Dependencies
+
+Assuming you already have [Homebrew](https://brew.sh) installed, the installation of dependencies is pretty straightforward and will generate the `xmr-stak` binary in the `bin/` directory.
+
+### For NVIDIA GPUs
+
+```shell
+brew tap caskroom/drivers && \
```
Comment (psychocrypt) [inline doc/compile_MacOS.md]: please remove `&& \` the most users will ignore the `\` or think they can copy and execute it line by line

File: doc/compile_MacOS.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+[More options](compile.md#nvidia-build-options) are available.
+
+### For AMD GPUs
+
+> 🖐 We need help with AMD GPU compilation instructions. Please submit a PR if you managed to install [AMD APP SDK](http://developer.amd.com/amd-accelerated-parallel-processing-app-sdk/) and to compile `xmr-stak` on MacOS.
+
+### For CPU-only mining
+
+```shell
+brew install hwloc libmicrohttpd gcc openssl cmake && \
```
Comment (psychocrypt) [inline doc/compile_MacOS.md]: please remove `&& \`

File: doc/compile_MacOS.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+cmake . -DOPENSSL_ROOT_DIR=/usr/local/opt/openssl -DCUDA_ENABLE=OFF -DOpenCL_ENABLE=OFF && \
+make install
+```
+
+[More options](compile.md#cpu-build-options) are available.
```
Comment (psychocrypt) [inline doc/compile_MacOS.md]: please change `More options` to `All available CMake options`

**Comment by psychocrypt** [comment]
please rebase you PR against the dev branch after #179 is merged. This will fix the windows CI tests with appvoyer.

**Comment by michaelbaudino** [comment]
Re-posted against `dev` as #204, sorry about not reading the `CONTRIBUTING` file 😖

**Comment by fireice-uk** [comment]
Closing. Thanks for contributing!

**Pull request closed without merging**